test/routes/actions.test.js

```diff
@@ -125,4 +159,102 @@ describe('routes > actions', () => {
+
+            const {
+              send, response, model, implementation,
+            } = await callLoadHook('oops');
+            expect(implementation.ResourceGetter)
+              .toHaveBeenNthCalledWith(1, model, { recordId: 1 });
+            expect(response.status).toHaveBeenNthCalledWith(1, 500);
+            expect(send).toHaveBeenNthCalledWith(1, { message: 'load must be a function' });
+          });
+
+          it('should fail with message when action.hooks.load does not return an object', async () => {
+            expect.assertions(4);
+
+            const load = jest.fn();
+            const {
+              send, response, model, implementation,
+            } = await callLoadHook(load);
+
+            expect(implementation.ResourceGetter)
+              .toHaveBeenNthCalledWith(1, model, { recordId: 1 });
+            expect(response.status).toHaveBeenNthCalledWith(1, 500);
+            expect(load).toHaveBeenNthCalledWith(1, {
+              fields: {
+                'invoice number': { field: 'invoice number', type: 'String', value: null },
+              },
+              record: { id: 1, name: 'Jane' },
+            });
+            expect(send).toHaveBeenNthCalledWith(1, { message: 'load hook must return an object' });
+          });
+
+          it('should fail with message when action.hooks.load returned fields are not consistent', async () => {
+            expect.assertions(2);
+
+            const newFields = {
+              'MODIFIED invoice number': { field: 'MODIFIED invoice number', type: 'String', value: null },
+            };
+            const load = jest.fn(() => newFields);
+            const {
+              send, response,
+            } = await callLoadHook(load);
+
+            expect(response.status).toHaveBeenNthCalledWith(1, 500);
+            expect(send).toHaveBeenNthCalledWith(1, { message: 'fields must be unchanged (no addition nor deletion allowed)' });
+          });
+        });
+        describe('when action.hooks.load is valid', () => {
+          it('should respond success with the updated fields', async () => {
+            expect.assertions(4);
+
+            const newFields = {
+              'invoice number': { field: 'invoice number', type: 'String', value: 'hello from load' },
+            };
+
+            const load = jest.fn(() => newFields);
+            const {
+              send, response, model, implementation,
+            } = await callLoadHook(load);
+
+            expect(implementation.ResourceGetter)
+              .toHaveBeenNthCalledWith(1, model, { recordId: 1 });
+            expect(response.status).toHaveBeenNthCalledWith(1, 200);
+            expect(load).toHaveBeenNthCalledWith(1, {
+              fields: {
+                'invoice number': { field: 'invoice number', type: 'String', value: null },
+              },
+              record: { id: 1, name: 'Jane' },
+            });
+            expect(send).toHaveBeenNthCalledWith(1, { fields: newFields });
```

Comment:
It should be:
```
expect(send).toHaveBeenNthCalledWith(1, { fields: [{ field: 'invoice number', type: 'String', value: 'hello from load' }] });
```

Otherwise, we will lose the position in the frontend.

Reply:
Fixed